atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPool.java

```diff
@@ -446,16 +446,32 @@ private void refreshTokenRanges() {
                 }
             };
 
+    public <V, K extends Exception> V runWithRetry(
+            FunctionCheckedException<Cassandra.Client, V, K> fn,
+            boolean printHost) throws K {
+        return runWithRetryOnHost(getRandomGoodHost().getHost(), fn, printHost);
+    }
+
     public <V, K extends Exception> V runWithRetry(FunctionCheckedException<Cassandra.Client, V, K> fn) throws K {
-        return runWithRetryOnHost(getRandomGoodHost().getHost(), fn);
+        return runWithRetryOnHost(getRandomGoodHost().getHost(), fn, false);
     }
 
     public <V, K extends Exception> V runWithRetryOnHost(
             InetSocketAddress specifiedHost,
             FunctionCheckedException<Cassandra.Client, V, K> fn) throws K {
+        return runWithRetryOnHost(specifiedHost, fn, false);
+    }
+
+    private <V, K extends Exception> V runWithRetryOnHost(
+            InetSocketAddress specifiedHost,
+            FunctionCheckedException<Cassandra.Client, V, K> fn,
+            boolean printHost) throws K {
         int numTries = 0;
         boolean shouldRetryOnDifferentHost = false;
         while (true) {
+            if (printHost) {
+                log.trace("Running function on host {}.", specifiedHost.getHostName());
```

Comment:
Should we remove the `printHost` param and just use the following (should be cheap enough if trace is disabled)? Note that I switch to `InetSocketAddress.getHostString()` to avoid blocking on DNS lookup, see  https://docs.oracle.com/javase/8/docs/api/java/net/InetSocketAddress.html#getHostString--

`log.trace("Running function on host {}.", specifiedHost.getHostString());`